### PR TITLE
spark: update everything

### PR DESCRIPTION
**Pull request by matz-e**
Includes updates to Arrow and Spark, and will enable dataset support in
Arrow to switch away from deprecated metadata reading.

New patch for Spark daemon launch should handle startup better and
actually fail when launching the Spark workers/master fails.

See also FUNCZ-310, FUNCZ-312.

### CODE REVIEW

File: bluebrain/repo-patches/packages/py-pyarrow/package.py [hunk 1]
```diff
@@ -5,10 +5,23 @@
 class PyPyarrow(BuiltinPyPyarrow):
     __doc__ = BuiltinPyPyarrow.__doc__
 
+    pypi = 'pyarrow/pyarrow-9.0.0.tar.gz'
+
+    version('9.0.0', sha256='7fb02bebc13ab55573d1ae9bb5002a6d20ba767bf8569b52fce5301d42495ab7')
```
Comment (GianlucaFicarelli) [inline bluebrain/repo-patches/packages/py-pyarrow/package.py:10]: FYI, with pyarrow 9.0.0 we had issues in the CI, when the CPU id 0 is not available. See https://issues.apache.org/jira/browse/ARROW-17319
cc @mgeplf

Comment (matz-e) [reply bluebrain/repo-patches/packages/py-pyarrow/package.py:10]: Hm, good point. Did you observe that with 8.0.0? I could just mark that as `preferred` for the time being.

Comment (mgeplf) [reply bluebrain/repo-patches/packages/py-pyarrow/package.py:10]: @matz-e how does pyarrow pull in it's `aws-c-*` dependencies?

Comment (mgeplf) [reply bluebrain/repo-patches/packages/py-pyarrow/package.py:10]: 8 wheels seemed to be fine, but I don't think we tracked when they started using the `aws` stuff.

Comment (matz-e) [reply bluebrain/repo-patches/packages/py-pyarrow/package.py:10]: @mgeplf it does not pull in any dependencies on AWS, we also build without S3 support. So I guess this should be fine? I'll try some reproducer on BB5

Comment (matz-e) [reply bluebrain/repo-patches/packages/py-pyarrow/package.py:10]: Hm, don't see this with py3.9 using a virtualenv with a `pyarrow` wheel, either.

Comment (matz-e) [reply bluebrain/repo-patches/packages/py-pyarrow/package.py:10]: OK, so could confirm the issue with a PYPI provided `pyarrow`. As we don't build with any AWS components, `py-pyarrow@9.0.0` is fine on BlueBrain5, though.

**Review by sergiorg-hpc — APPROVED** [review]

**Pull request merged**